-- @testpoint: 仅带出参，出参类型为DATETIME,执行INSERT INTO的存储过程，合理报错
drop table if exists TEST_001;
SQL SUCCESS
CREATE TABLE TEST_001(
  T1 INT,
  T2 INTEGER NOT NULL,
  T3 BIGINT,
  T4 NUMBER DEFAULT 0.2332,
  T5 NUMBER(12,2),
  T6 NUMBER(12,6),
  T7 BINARY_DOUBLE,
  T8 DECIMAL,
  T9 DECIMAL(8,2),
  T10 DECIMAL(8,4),
  T11 REAL,
  T12 CHAR(4000),
  T13 CHAR(100),
  T14 VARCHAR(4000),
  T15 VARCHAR(100),
  T16 VARCHAR2(4000),
  T17 NUMERIC,

  T19 DATE,
  T20 TIMESTAMP,
  T21 TIMESTAMP(6),
  T22 BOOL
) ;
SQL SUCCESS
create unique index  indx_out_param_t1 on TEST_001(T1);
SQL SUCCESS
create index indx_out_param_t2 on TEST_001(T2,T17,T20);
SQL SUCCESS
SQL SUCCESS
SQL SUCCESS
SQL SUCCESS
SQL SUCCESS
SQL SUCCESS
--仅带出参的存储过程测试（测试出参类型为DATETIME,执行INSERT INTO的存储过程）
CREATE OR REPLACE PROCEDURE PROC_OUT_PARAM_0041(P1 OUT DATETIME) 
AS
BEGIN

EXCEPTION
WHEN NO_DATA_FOUND THEN raise info 'NO_DATA_FOUND';
END;
/
?.*ERROR: type datetime does not exist
--调用存储过程
BEGIN 

END;
/
?.*ERROR: duplicate key value violates unique constraint "indx_out_param_t1"
  Detail: Key (t1)=(16) already exists.
PL/pgSQL function inline_code_block line 3 at SQL statement
DROP TABLE TEST_001;
SQL SUCCESS
drop PROCEDURE PROC_OUT_PARAM_0041;
?.*
drop table if exists test_001;
SQL SUCCESS